### PR TITLE
Unrename scales to functions

### PR DESCRIPTION
**Pull request by lucaswoj**
... and rename the function types in the reference JSON

cc @jfirebaugh @ansis 

https://mapbox.slack.com/archives/gl/p1434660255010066

### CODE REVIEW

**Comment by lucaswoj** [comment]
@jfirebaugh this is also ready for :eyes:

**Comment by jfirebaugh** [comment]
`s/continuious/continuous/g`

**Comment by lucaswoj** [comment]
Ah! Too much copy paste.

File: reference/v8.json [hunk 1]
```diff
@@ -826,32 +786,31 @@
       "mix"
     ]
   },
-  "function": {
+  "function-v7": {
```
Comment (jfirebaugh) [inline reference/v8.json]: Is it necessary to keep this?

Comment (lucaswoj) [inline reference/v8.json]: Depends. Do we still want the ability to validate v7 styles?

File: lib/validate/parsed.js [hunk 2]
```diff
@@ -50,12 +50,12 @@ module.exports = function(style, reference) {
 
         // Functions
         if (reference.$version < 8 && spec.function && type === 'object') {
```
Comment (jfirebaugh) [inline lib/validate/parsed.js]:
```
if (spec.function && type === 'object') {
    return validate[reference.$version >= 8 ? 'function' : 'function-v7'](key, val, spec);
}
```

File: reference/v8.json (hunk 1 above)
Comment (jfirebaugh) [inline reference/v8.json]: Yes, but when validating a v7 style the `reference` that will be passed in [here](https://github.com/mapbox/mapbox-gl-style-spec/blob/696fda69a24f59ef578db47558840404712b5895/lib/validate/parsed.js#L6) will come from `reference/v7.json`. [See here](https://github.com/mapbox/mapbox-gl-style-spec/blob/696fda69a24f59ef578db47558840404712b5895/lib/validate.js#L41).

Comment (lucaswoj) [inline reference/v8.json]: Of course! How silly of me.

**Comment by lucaswoj** [comment]
@jfirebaugh :ship:?

**Comment by jfirebaugh** [comment]
:ship:

**Pull request merged**